packages/common/src/converter/data-converter.ts

```diff
@@ -182,8 +191,22 @@ export function mapToPayloadsSync<K extends string>(
   ) as Record<K, Payload>;
 }
 
-export const defaultDataConverter = new CompositeDataConverter(
-  new UndefinedPayloadConverter(),
-  new BinaryPayloadConverter(),
-  new JsonPayloadConverter()
-);
+export interface DefaultDataConverterOptions {
+  root?: Record<string, unknown>;
+}
+
+export class DefaultDataConverter extends CompositeDataConverter {
+  constructor({ root }: DefaultDataConverterOptions = {}) {
+    // Match the order used in other SDKs
+    // Go SDK: https://github.com/temporalio/sdk-go/blob/5e5645f0c550dcf717c095ae32c76a7087d2e985/converter/default_data_converter.go#L28
+    super(
+      new UndefinedPayloadConverter(),
+      new BinaryPayloadConverter(),
+      new ProtobufJsonPayloadConverter(root),
+      new ProtobufBinaryPayloadConverter(root),
```

Comment:
Once Java add the binary converter to the default converter we should change the order to binary -> json.
Let's add a todo or issue.